mimic/model/nova_objects.py

```diff
@@ -927,6 +940,67 @@ def request_action(self, http_action_request, server_id, absolutize_url):
         else:
             return dumps(bad_request("There is no such action currently supported", http_action_request))
 
+    def _create_list_of_flavors(self):
+        """
+        Creates a list of flavors and stores them in the flavors store
+        """
+        flavors = [StandardFlavor, ComputeFlavor, Performance1Flavor, OnMetalFlavor,
+                   Performance2Flavor, MemoryFlavor, IOFlavor, GeneralFlavor]
```

Comment:
Yeah, I don't like the fact that a method called `_generate_<something>` actually both generates the things and appends them to the flavors store.  This means tighter coupling, and it's harder to read it and understand its correctness.  I'd much rather that we have a *function* that does the generation, and is stateless, and then a method that does the adding.

There are only two usages of `self` in `_generate_flavors_info`; the first is to test to see if the flavor is already available or not, and then silently refusing to add a duplicate if it is.  The second is to append to `flavors_store`.

As to the first, it seems to me that this should be an error, because what if the flavor ID is the same but the flavor data is different?  Allowing that to pass silently could possibly create a very confusing situation where some code added a flavor but then it was different in the actual flavors list.  The second `self` reference could be replaced with `yield flavor` and then making this a function.

I'm okay with the specific detail of passing in a list of classes, but "rackspace" should be in the name somewhere given that these flavors are rackspace-specific.